base/inference.jl

```diff
@@ -1590,7 +1618,7 @@ function typeinf_edge(method::Method, atypes::ANY, sparams::SimpleVector, caller
             end
         end
     end
-    frame = typeinf_frame(code, true, true, caller)
+    frame = typeinf_frame(code, caller, InferenceParams(caller.params; cached=true))
```

Comment:
this is a hot path, it would be nice to avoid the kwargs dispatch and allocation here